stable/yugabyte/templates/service.yaml

```diff
@@ -145,12 +145,16 @@ spec:
     {{- range $index := until (int ($storageInfo.count )) }}
     - metadata:
         name: {{ $root.Values.oldNamingStyle | ternary (printf "datadir%d" $index) (printf "%s%d" (include "yugabyte.volume_name" $root) $index) }}
+        {{- if $root.Values.legacyVolumeClaimAnnotations }}
+        annotations:
+          volume.beta.kubernetes.io/storage-class: {{ $storageInfo.storageClass | quote }}
+        {{- end }}
         labels:
           {{- include "yugabyte.labels" $root | indent 10 }}
       spec:
         accessModes:
           - "ReadWriteOnce"
-        {{- if $storageInfo.storageClass }}
+        {{- if or $root.Values.legacyVolumeClaimAnnotations $storageInfo.storageClass }}
```

Comment:
I think we don't need this change?
Because old installations will have storageClass set to `standard`, if they have specifically set it to a blank string (I don't think anyone will be there but still), then we are breaking their setup here if they set `legacyVolumeClaimAnnotations`.

Reply:
Ref - https://kubernetes.io/docs/concepts/storage/persistent-volumes/#reserving-a-persistentvolume